pkg/controllers/dynakube/otelc/statefulset/reconciler.go

```diff
@@ -41,7 +43,9 @@ func NewReconciler(clt client.Client,
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context) error {
-	if !r.dk.IsExtensionsEnabled() {
+	if r.dk.IsExtensionsEnabled() || r.dk.TelemetryService().IsEnabled() {
```

Comment:
Should we put this check into a common function `dk.IsOpenTelemetryCollectorRequired()`?

Reply:
we use it only in single place, AFAIK.